packages/bitcore-lib/lib/script/interpreter.js

```diff
@@ -70,26 +72,133 @@ Interpreter.prototype.verifyWitnessProgram = function(version, program, witness,
       scriptPubKey.add(Opcode.OP_CHECKSIG);
 
       stack = witness;
-
+      return this.executeWitnessScript(scriptPubKey, stack, Signature.Version.WITNESS_V0, satoshis, flags);
     } else {
       this.errstr = 'SCRIPT_ERR_WITNESS_PROGRAM_WRONG_LENGTH';
       return false;
     }
+  } else if (version === 1 && program.length == Interpreter.WITNESS_V1_TAPROOT_SIZE && !isP2SH) {
+    const execdata = { annexPresent: false };
+    // BIP341 Taproot: 32-byte non-P2SH witness v1 program (which encodes a P2C-tweaked pubkey)
+    if (!(flags & Interpreter.SCRIPT_VERIFY_TAPROOT)) {
+      return true;
+    }
+    stack = Array.from(witness);
+    if (stack.length == 0) {
+      this.errstr = 'SCRIPT_ERR_WITNESS_PROGRAM_WITNESS_EMPTY';
+      return false;
+    }
+    if (stack.length >= 2 && stack[stack.length - 1].length && stack[stack.length - 1][0] === Script.ANNEX_TAG) {
+      // Drop annex (this is non-standard; see IsWitnessStandard)
+      const annex = stack.pop();
+      const annexWriter = new BufferWriter();
+      annexWriter.writeVarintNum(annex.length);
+      annexWriter.write(annex);
+      execdata.annexHash = Hash.sha256(annexWriter.toBuffer());
+      execdata.annexPresent = true;
+    }
+    execdata.annexInit = true;
+    if (stack.length === 1) {
```

Comment:
This could simply be:
`
if (stack.length === 1) {
  return this.checkSchnorrSignature(...)
}
`